typescript/web/src/components/members/new-member-modal.tsx

```diff
@@ -1,35 +1,39 @@
-import { useState, ChangeEvent } from "react";
 import {
-  Flex,
   Alert,
-  AlertIcon,
   AlertDescription,
-  VStack,
-  Text,
-  Textarea,
+  AlertIcon,
+  Button,
+  Flex,
+  Heading,
+  Link,
   Modal,
-  ModalOverlay,
-  ModalContent,
+  ModalBody,
   ModalCloseButton,
+  ModalContent,
   ModalFooter,
-  Button,
-  Heading,
   ModalHeader,
-  ModalBody,
+  ModalOverlay,
+  Text,
+  Textarea,
   useColorModeValue as mode,
-  Link,
   useToast,
+  VStack,
 } from "@chakra-ui/react";
+import { InvitationResult, MembershipRole } from "@labelflow/graphql-types";
+import { isEmpty } from "lodash/fp";
 import { useRouter } from "next/router";
-import { MembershipRole, InvitationResult } from "@labelflow/graphql-types";
-
+import { ChangeEvent, useState } from "react";
+import { validateEmail } from "../../utils/validate-email";
 import { RoleSelection } from "./role-selection";
 import { InviteMember } from "./types";
-import { validateEmail } from "../../utils/validate-email";
 
 const maxNumberOfEmails = 20;
 type EmailStatuses = Record<InvitationResult, string[]>;
 
+const parseEmails = (text: string): string[] => {
+  return text.split(/[,;\s]/).filter((email) => !isEmpty(email));
+};
+
```

Comment:
@nicOwlas I've moved the corresponding code in a `parseEmails` function which could be eventually unit-tested in the future (so that we could cover each case and new bugs related with it).